### PR TITLE
Bump Stylelint dependencies to latest v14.x compatible releases

### PR DESCRIPTION
**Pull request by ntwb**
## What?

Bumps `stylelint` from v14.2 to v14.16.1
- https://github.com/stylelint/stylelint/releases/tag/14.16.1
- This is the latest in the v14.x branch, before the v15.x rule removals

Bumps `stylelint-config-recommended` from v6.0.0 to v9.0.0
- https://github.com/stylelint/stylelint-config-recommended/releases/tag/9.0.0
- This release is the latest release compatible with Stylelint v14.x

Bumps `stylelint-config-recommended-scss` from v5.0.2 to v8.0.0
- https://github.com/stylelint-scss/stylelint-config-recommended-scss/releases/tag/v8.0.0
- This release is the latest release compatible with Stylelint v14.x

Dependencies changes remove `^` semver caret, because Lerna is not a fan of installing specific versions, it's moot as each of these dependency bumps are the latest andlast releases in each of these _major_ versions of each package

## Why?

With the Gutenberg project blocked from using Stylelint v15 because of the removal of "stylistic" rules this PR seeks to update Stylelint dependencies to latest compatible with Stylelint v14

## How?

Packages were updated using the guide here: https://github.com/WordPress/gutenberg/blob/trunk/packages/README.md

Each package was first removed from each package from the `dependencies` section of the package `package.json` file

Each package upgrade was then added using Lerna v5.x
- `lerna add stylelint-config-recommended@9 packages/stylelint-config/`
- `lerna add stylelint-config-recommended-scss@8 packages/stylelint-config/`
- `lerna add stylelint@14.16.1 packages/scripts/`

## Testing Instructions

1. Checkout the `bump-stylelint-config-recommended-packages` branch
2. Run `npm install`
3. Run `npm run build`
4. Run `npm run lint:css`

## Notes 

See this issue comment in relation to why Stylelint is not being bumped to v15 at this stage https://github.com/WordPress/gutenberg/pull/50336#issuecomment-1554215986
- tl;dr: Stylelint v15 removes "stylistic" rules (formatting) in favour of using Prettier for "formatting" and retaining only "syntax" rules that affect "functionality" rather than "style"

## CI Failures 

- ✅ Getting closer,some deps were not correctly bootstrapped with Lerna, that's fixed
- 🚧 Looking into `TypeError [ERR_INVALID_URL_SCHEME]: The URL must be of scheme file` issue
  - [Stylelint 14.5.1](https://github.com/stylelint/stylelint/issues/5919) Resolved this issue, so it shouldn't be an issue here

### CODE REVIEW

**Comment by gziolo** [comment]
There is now another more up-to-date PR from @mikeybinns that is fully functional, ready for review and testing:
- https://github.com/WordPress/gutenberg/pull/64828

Thank you so much for your contribution!
